src/data/fin.lean

```diff
@@ -986,6 +988,8 @@ pred_above (last n) i
 
 @[simp] lemma cast_pred_zero : cast_pred (0 : fin (n + 2)) = 0 := rfl
 
+@[simp] lemma cast_pred_one : cast_pred (1 : fin (n + 3)) = 1 := rfl
```

Comment:
Did you try this for `n+2`? It seems like it might just work.

Can we have similar lemmas for `bit0` and `bit1`? Or are there complications?

Reply:
In `fin (n + 2)` the right hand side is `fin (n + 1)`, and that includes `fin 1` where `0 = 1` and this lemma is false

Reply:
bit0 and bit1 are a nice idea, and something I might try another time.

Reply:
```suggestion
@[simp] lemma cast_pred_one : cast_pred (1 : fin (n + 2)) = 1 := 
by { cases n, apply subsingleton.elim, refl }
```
This works just fine. Unless it's really important for you that `dsimp` can apply it, but then I think you want a separate lemma for `dsimp`.